File: full_case_study_solns/_f_planar_vtol/python_old/hw7/VTOLController.py
import numpy as np
import sys
sys.path.append('..')  # add parent directory
import VTOLParam as P
import VTOLParamHW7 as P7
from PDControl import PDControl

class VTOLController:
    ''' 
        This class inherits other controllers in order to organize multiple controllers.
    '''

    def __init__(self):
        # Instantiates the SS_ctrl object
        self.hCtrl = PDControl(P7.kp_h, P7.kd_h, P.fmax, P.beta, P.Ts)

    def u(self, r, y):
        z_r = float(r[0])
        h_r = float(r[1])
        z = y[0]
        h = y[1]
        theta = y[2]
        F_tilde = self.hCtrl.PD(h_r, h, flag=False)
        F = F_tilde + P.Fe
        tau = 0.0
        return np.matrix([[F], [tau]])

File: full_case_study_solns/_f_planar_vtol/matlab_old/hw7/VTOLController.m
classdef VTOLController
    %----------------------------
    properties
        hCtrl
        Fe
    end
    %----------------------------
    methods
        %----------------------------
        function self = VTOLController(P)
            self.Fe = P.Fe;
            % Instantiates the PD control objects
            self.hCtrl = PDControl(P.kp_h, P.kd_h, P.fmax, P.beta, P.Ts);
        end
        %----------------------------
        function out = u(self, r, y)
            z_r = r(1);
            h_r = r(2);
            z = y(1);
            h = y(2);
            theta = y(3);
            % the force applied to the cart comes from the inner loop PD control
            F_tilde = self.hCtrl.PD(h_r, h, false);
            % total force
            F = F_tilde + self.Fe;
            tau = 0;
            out = [F; tau];
        end
    end
end